Assets/Scripts/Battlefield/TurnMechanism/TurnManager.cs

```diff
@@ -131,36 +132,26 @@ void Update()
                 }
             }
 
-            if (enemies.Count == 0)
+            if (enemies.Count == 0 || playerUnits.Count == 0)
             {
-                WinCondition();
-            }
-            if (playerUnits.Count == 0)
-            {
-                LoseCondition();
+                EndBattle(enemies.Count == 0);
             }
             CheckStatusForAll();
-
         }
 
-        public void WinCondition()
+        public void EndBattle(bool playerWin)
         {
+            if (hasLoadedEndScene)
+            {
+                return;

```

Comment:
Instead of exiting the method early, remove the call to the method in update

Reply:
It's fine if you don't want to change now, I will change a lot of the conditions when I add getting an item from battle. It will be then UI based to return to strategy